master/buildbot/test/unit/test_push_secret_to_worker.py

```diff
@@ -0,0 +1,72 @@
+# This file is part of Buildbot.  Buildbot is free software: you can
+# redistribute it and/or modify it under the terms of the GNU General Public
+# License as published by the Free Software Foundation, version 2.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+# FOR A PARTICULAR PURPOSE.  See the GNU General Public License for more
+# details.
+#
+# You should have received a copy of the GNU General Public License along with
+# this program; if not, write to the Free Software Foundation, Inc., 51
+# Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# Copyright Buildbot Team Members
+
+from __future__ import absolute_import
+from __future__ import print_function
+
+import os
+
+from twisted.python.filepath import FilePath
+from twisted.trial import unittest
+
+from buildbot.process.results import SUCCESS
+from buildbot.steps.push_secret_to_worker import PushSecretToWorker
+from buildbot.steps.push_secret_to_worker import RemoveWorkerFileSecret
+from buildbot.test.util import config as configmixin
+from buildbot.test.util import steps
+
+
+class TestPushSecretToWorkerStep(steps.BuildStepMixin, unittest.TestCase, configmixin.ConfigErrorsMixin):
+
+    def createTempDir(self, dirname):
+        tempdir = FilePath(self.mktemp())
+        tempdir.createDirectory()
+        return tempdir.path
+
+    def setUp(self):
+        self.temp_path = self.createTempDir("tempdir")
+
+    def testPushSecretToWorkerStepSuccess(self):
+        self.setupStep(PushSecretToWorker([(os.path.join(self.temp_path, "pathA"), "something")]))
+        self.expected_remote_commands = ""
```

Comment:
you should actually test that remote commands are created which create the files